File: sdks/python/roost/resources/deployments.py
"""``roost.deployments`` — per-roost rollout history + detail."""

from __future__ import annotations

from collections.abc import AsyncIterator
from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from roost.client import RoostClient


class Deployments:
    def __init__(self, client: "RoostClient") -> None:
        self._client = client

    async def list_page(
        self,
        roost_id: str,
        *,
        site_id: str,
        page_size: int = 20,
        page_token: str | None = None,
        cursor: str | None = None,
    ) -> tuple[list[dict[str, Any]], str]:
        """One page of deployment rollouts. Returns ``(rollouts, next_page_token)``."""
        token = page_token if page_token is not None else cursor
        resp = await self._client.request(
            f"/api/roosts/{roost_id}/deployments",
            query={"siteId": site_id, "page_size": page_size, "page_token": token},
        )
        data = resp.data if isinstance(resp.data, dict) else {}
        rollouts = [r for r in data.get("rollouts", []) or data.get("items", []) if isinstance(r, dict)]
        next_token = data.get("nextPageToken") or data.get("next_page_token") or ""
        return rollouts, str(next_token)

    async def list(
        self,
        roost_id: str,
        *,
        site_id: str,
        page_size: int = 20,
        page_token: str | None = None,
        cursor: str | None = None,
    ) -> AsyncIterator[dict[str, Any]]:
        token: str | None = page_token if page_token is not None else cursor
        while True:
            rows, token = await self.list_page(
                roost_id,
                site_id=site_id,
                page_size=page_size,
                page_token=token,
            )
            for r in rows:
                yield r
            if not token:
                return

    async def get(
        self, roost_id: str, rollout_id: str, *, site_id: str
    ) -> dict[str, Any]:
        resp = await self._client.request(
            f"/api/roosts/{roost_id}/deployments/{rollout_id}",
            query={"siteId": site_id},
        )
        return resp.data if isinstance(resp.data, dict) else {}


__all__ = ["Deployments"]

File: sdks/node/src/resources/deployments.ts
import type { OwletteClient } from '../lib/client';

export interface ListDeploymentsOptions {
  siteId: string;
  /** Deprecated alias for pageSize; use pageSize for new code. */
  limit?: number;
  /** Deprecated alias for pageToken; use pageToken for new code. */
  cursor?: string;
  pageSize?: number;
  pageToken?: string;
}

export class Deployments {
  constructor(private readonly client: OwletteClient) {}

  async list(
    roostId: string,
    opts: ListDeploymentsOptions,
  ): Promise<{ rollouts: Array<Record<string, unknown>>; nextPageToken: string }> {
    const res = await this.client.request<{
      rollouts: Array<Record<string, unknown>>;
      nextPageToken: string;
    }>(`/api/roosts/${encodeURIComponent(roostId)}/deployments`, {
      query: {
        siteId: opts.siteId,
        page_size: opts.pageSize ?? opts.limit,
        page_token: opts.pageToken ?? opts.cursor,
      },
    });
    return res.data;
  }

  async get(
    roostId: string,
    rolloutId: string,
    opts: { siteId: string },
  ): Promise<Record<string, unknown>> {
    const res = await this.client.request<Record<string, unknown>>(
      `/api/roosts/${encodeURIComponent(roostId)}/deployments/${encodeURIComponent(rolloutId)}`,
      { query: { siteId: opts.siteId } },
    );
    return res.data;
  }
}
